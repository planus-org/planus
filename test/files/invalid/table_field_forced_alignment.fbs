table Align1 {
  value: uint16 (force_align: 1);
}

table Align2 {
  value: uint32 (force_align: 2);
}

table Align3 {
  value: uint8 (force_align: 3);
}

table Align4 {
  value: uint64 (force_align: 4);
}

struct Inner (force_align: 8) {
  value: uint8;
}

table Align8 {
  value: Inner (force_align: 4);
}
