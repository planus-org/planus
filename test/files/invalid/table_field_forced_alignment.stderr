error: Alignment must be a power of two
   ┌─ test/files/invalid/table_field_forced_alignment.fbs:10:17
   │
10 │   value: uint8 (force_align: 3);
   │                 ^^^^^^^^^^^^^^

error: Alignment of table field cannot be forced to lower
  ┌─ test/files/invalid/table_field_forced_alignment.fbs:2:18
  │
2 │   value: uint16 (force_align: 1);
  │          ------  ^^^^^^^^^^^^^^ This attribute tries to force the alignment of the field
  │          │        
  │          However the minimum alignment of this type is 2

error: Alignment of table field cannot be forced to lower
  ┌─ test/files/invalid/table_field_forced_alignment.fbs:6:18
  │
6 │   value: uint32 (force_align: 2);
  │          ------  ^^^^^^^^^^^^^^ This attribute tries to force the alignment of the field
  │          │        
  │          However the minimum alignment of this type is 4

error: Alignment of table field cannot be forced to lower
   ┌─ test/files/invalid/table_field_forced_alignment.fbs:14:18
   │
14 │   value: uint64 (force_align: 4);
   │          ------  ^^^^^^^^^^^^^^ This attribute tries to force the alignment of the field
   │          │        
   │          However the minimum alignment of this type is 8

error: Alignment of table field cannot be forced to lower
   ┌─ test/files/invalid/table_field_forced_alignment.fbs:22:17
   │
22 │   value: Inner (force_align: 4);
   │          -----  ^^^^^^^^^^^^^^ This attribute tries to force the alignment of the field
   │          │       
   │          However the minimum alignment of this type is 8

